动转静不支持broadcast_to算子

### bug描述 Describe the Bug

在动转静时，遇到以下代码报错
```
        time_context = time_context_first_timestep[None, :].broadcast_to(
            [height * width, batch_size, 1, time_context.shape[-1]]
        )
```

具体报错为
```
    error_data.raise_new_exception()
  File "/root/miniconda3/envs/paddle-develop/lib/python3.9/site-packages/paddle/jit/dy2static/error.py", line 452, in raise_new_exception
    raise new_exception from None
AssertionError: In transformed code:

    File "/root/project/paddlemix/upgrade_ppdiffuser0240/PaddleMIX/ppdiffusers/ppdiffusers/models/unet_spatio_temporal_condition.py", line 446, in forward
        for downsample_block in self.down_blocks:
    File "/root/project/paddlemix/upgrade_ppdiffuser0240/PaddleMIX/ppdiffusers/ppdiffusers/models/unet_spatio_temporal_condition.py", line 447, in forward
        if hasattr(downsample_block, "has_cross_attention") and downsample_block.has_cross_attention:
    File "/root/project/paddlemix/upgrade_ppdiffuser0240/PaddleMIX/ppdiffusers/ppdiffusers/models/unet_spatio_temporal_condition.py", line 448, in forward
        sample, res_samples = downsample_block(
    File "/root/project/paddlemix/upgrade_ppdiffuser0240/PaddleMIX/ppdiffusers/ppdiffusers/models/unet_3d_blocks.py", line 2129, in forward
        for resnet, attn in blocks:
    File "/root/project/paddlemix/upgrade_ppdiffuser0240/PaddleMIX/ppdiffusers/ppdiffusers/models/unet_3d_blocks.py", line 2130, in forward
        if self.training and self.gradient_checkpointing and not hidden_states.stop_gradient:  # TODO
    File "/root/project/paddlemix/upgrade_ppdiffuser0240/PaddleMIX/ppdiffusers/ppdiffusers/models/unet_3d_blocks.py", line 2162, in forward
        hidden_states = attn(
    File "/root/project/paddlemix/upgrade_ppdiffuser0240/PaddleMIX/ppdiffusers/ppdiffusers/models/transformer_temporal.py", line 327, in forward
        print("batch_size", batch_size)
        print("time_context.shape[-1]", time_context.shape[-1])
        time_context = time_context_first_timestep[None, :].broadcast_to(
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
            [height * width, batch_size, 1, time_context.shape[-1]]
        )

    File "/root/miniconda3/envs/paddle-develop/lib/python3.9/site-packages/paddle/tensor/manipulation.py", line 4146, in broadcast_to
        assert (

    AssertionError: Elements in shape must be 1-D Tensors or integers.var tmp_25 : LOD_TENSOR.shape().dtype(int32).stop_gradient(True)
```

替换为expand后，问题解决
```
        time_context = time_context_first_timestep[None, :].expand(
            [height * width, batch_size, 1, time_context.shape[-1]]
        )
```

总结：
paddle动转静貌似不支持broadcast_to算子?

环境信息：
paddlenlp                 2.6.2
paddlepaddle-gpu          2.6.0
fastdeploy-gpu-python     1.0.7
ppdiffusers               0.24.0

### 其他补充信息 Additional Supplementary Information

_No response_

--- a/python/paddle/tensor/manipulation.py
+++ b/python/paddle/tensor/manipulation.py
@@ -30,7 +30,6 @@ from ..base.data_feeder import (
 from ..base.framework import Variable, default_main_program
 from ..framework import (
     LayerHelper,
-    _current_expected_place,
     convert_np_dtype_to_dtype_,
     core,
     dygraph_only,
@@ -4125,89 +4124,7 @@ def broadcast_to(x, shape, name=None):
             [[1, 2, 3],
              [1, 2, 3]])
     """
-    if in_dynamic_mode():
-        return _C_ops.expand(x, shape)
-    elif in_pir_mode():
-        place = _current_expected_place()
-        if isinstance(shape, (list, tuple)):
-            if paddle.utils._contain_var(shape):
-                shape = paddle.utils.get_int_tensor_list(shape, place)
-        elif isinstance(shape, paddle.pir.OpResult):
-            shape.stop_gradient = True
-        else:
-            TypeError("Shape only supports OpReslut, or list, or tuple.")
-        return _C_ops.expand(x, shape)
-    else:
-        if isinstance(shape, Variable):
-            assert len(shape.shape) == 1, 'shape must be an 1-D Tensor.'
-        else:
-            type_tuple = (int, np.int32, np.int64)
-            for elem in shape:
-                if isinstance(elem, Variable):
-                    assert (
-                        len(elem.shape) == 1
-                    ), 'Elements in shape must be 1-D Tensors or integers.'
-                else:
-                    assert isinstance(
-                        elem, type_tuple
-                    ), 'Elements in shape must be 1-D Tensors or integers.'
-
-        check_variable_and_dtype(
-            x,
-            'x',
-            [
-                'bool',
-                'uint16',
-                'float16',
-                'float32',
-                'float64',
-                'int32',
-                'int64',
-            ],
-            'broadcast_to',
-        )
-        check_type(shape, 'shape', (list, tuple, Variable), 'broadcast_to')
-        if convert_dtype(x.dtype) == 'bool' and not x.stop_gradient:
-            raise ValueError(
-                "When the data type of input 'x' for broadcast_to is bool, "
-                "you must set its stop_gradient to be False by "
-                "some_var.stop_gradient = True, supporting "
-                "some_var as the input."
-            )
-
-        inputs = {"X": [x]}
-        attrs = {}
-
-        helper = LayerHelper('expand', **locals())
-
-        def get_attr_expand_shape(list_expand_shape):
-            attrs_expand_shape = []
-            for idx, shape in enumerate(list_expand_shape):
-                if isinstance(shape, Variable):
-                    attrs_expand_shape.append(-1)
-                else:
-                    attrs_expand_shape.append(shape)
-                    assert (
-                        shape > 0 or shape == -1
-                    ), "All elements in shape of broadcast_to must be positive or -1."
-            return attrs_expand_shape
-
-        if isinstance(shape, Variable):
-            shape.stop_gradient = True
-            inputs['Shape'] = shape
-        elif isinstance(shape, (list, tuple)):
-            attrs['shape'] = get_attr_expand_shape(shape)
-            if paddle.utils._contain_var(shape):
-                inputs[
-                    'expand_shapes_tensor'
-                ] = paddle.utils._convert_to_tensor_list(shape)
-
-        dtype = helper.input_dtype(input_param_name='x')
-        out = helper.create_variable_for_type_inference(dtype)
-        helper.append_op(
-            type='expand_v2', inputs=inputs, outputs={'Out': out}, attrs=attrs
-        )
-        return out
+    return expand(x, shape, name)
 
 
 def expand(x, shape, name=None):

--- a/python/paddle/tensor/manipulation.py
+++ b/python/paddle/tensor/manipulation.py
@@ -4176,7 +4176,7 @@ def expand(x, shape, name=None):
         return _C_ops.expand(x, shape)
     else:
         if isinstance(shape, Variable):
-            assert shape.numel() == 1, 'shape must be a Tensor with one element'
+            assert len(shape.shape()) == 1, 'shape must be a 1-D Tensor.'
         else:
             for elem in shape:
                 if isinstance(elem, Variable):

--- a/python/paddle/tensor/manipulation.py
+++ b/python/paddle/tensor/manipulation.py
@@ -4176,7 +4176,7 @@ def expand(x, shape, name=None):
         return _C_ops.expand(x, shape)
     else:
         if isinstance(shape, Variable):
-            assert len(shape.shape()) == 1, 'shape must be a 1-D Tensor.'
+            assert len(shape.shape) == 1, 'shape must be a 1-D Tensor.'
         else:
             for elem in shape:
                 if isinstance(elem, Variable):

